### ISSUE TITLE
Add build project before launch task 

### ISSUE THREAD

**Issue by hurricup** [Camelcade/Perl5-IDEA#2628]
If we have a `Makefile.PL` we need to run it with `make`

![image](https://user-images.githubusercontent.com/2811330/224527974-31de631f-6585-42e8-b34a-02d8f1459156.png)

**Comment by rwp0** [comment]
Actually we do `perl Makefile.PL`, and then run `make` on the generated `Makefile`.

And that's only for `ExtUtils-MakeMaker` distributions.

For `Module-Build` distributions we do:

```
perl Build.PL
./Build
./Build test
./Build install
```

instead.

The latter, being pure Perl, doesn't require the `make` utility (which is absent in Windows).

**Comment by hurricup** [comment]
Looks like there are several build systems we need to support: `ExtUtils::MakeMaker`, `Module::Build`, `Dist::Zilla` (which is just a proxy to the `MakeMaker` in terms of building as far as I understand, but still)

Also there is an issue with prove, see Perl-Toolchain-Gang/Test-Harness#117

**Comment by hurricup** [comment]
perl on windows may have `make` or `dmake`. Msys2 helps a lot with this.

**Comment by rwp0** [comment]
> `Dist::Zilla` (which is just a proxy to the `MakeMaker` in terms of building as far as I understand, but still)

Dist-Zilla can be a front-end for many things including [ExtUtils-MakeMaker](https://metacpan.org/pod/Dist::Zilla::Plugin::MakeMaker) and [Module-Build](https://metacpan.org/pod/Dist::Zilla::Plugin::ModuleBuild).

It just has a lot of built-in, and external plugins that can do almost anything imaginable.

**Comment by hurricup** [comment]
I expect a lot of problems with `Dist::Zilla` because it makes/runs tests in the build sandbox. Not sure how to handle that yet. Probably will need to map stuff like coverage/profiling/debugging

**Comment by rwp0** [comment]
> problems

Yes, it's not in the core, and has a lot of dependencies, `Moose` included.

So I think it might be better to start supporting ExtUtils-MakeMaker and Module-Build as the Perl build systems in the initial phase.

What I personally do is just create [run configurations](https://github.com/rwp-cpan/WWW-PiHole/tree/master/.idea/runConfigurations) for Dist-Zilla to be run in the Run window which is being almost always just enough for my use case.

So a plain menu with those for command could suffice for now:

```
dzil build
dzil clean
dzil install
dzil release
```
